modules/caddyhttp/reverseproxy/streaming.go

```diff
@@ -32,21 +32,20 @@ func (h Handler) handleUpgradeResponse(rw http.ResponseWriter, req *http.Request
 	reqUpType := upgradeType(req.Header)
 	resUpType := upgradeType(res.Header)
 	if reqUpType != resUpType {
-		// TODO: figure out our own error handling
-		// p.getErrorHandler()(rw, req, fmt.Errorf("backend tried to switch protocol %q when %q was requested", resUpType, reqUpType))
+		h.logger.Sugar().Errorf("backend tried to switch protocol %q when %q was requested", resUpType, reqUpType)
```

Comment:
Not a server error, so maybe DEBUG level

Reply:
Should I continue to use the sugar here cause it's using a formatted string?

Reply:
Hmmm well we don't use sugared logs anywhere else in Caddy so I dunno why we need to start here. I usually prefer to keep it consistently structured, even if individual fields require reflection to fill. (which is rare; as are these logs)